### PR TITLE
IBX-8726: `IsBookmarked` search criterion added

### PR DESCRIPTION
**Pull request by julitafalcondusza**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | (https://github.com/ibexa/core/pull/417)
| Versions      | 4.6+

`IsBookmarked` search criterion added

Previews:
- https://ez-systems-developer-documentation--2545.com.readthedocs.build/en/2545/search/criteria_reference/search_criteria_reference/#search-criteria The table now have `IsBookmarked` available for **Location Search** and **Filtering**.
- https://ez-systems-developer-documentation--2545.com.readthedocs.build/en/2545/search/criteria_reference/isbookmarked_criterion/

#### Checklist

- [x] Text renders correctly
- [x] Text has been checked with vale
- [x] Description metadata is up to date
- [x] Code samples are working

### CODE REVIEW

**Review by ciastektk — COMMENTED** [review]
REST API examples are missing: https://github.com/ibexa/rest/pull/123

**Review by ciastektk — APPROVED** [review]

File: docs/search/criteria_reference/isbookmarked_criterion.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+description: IsBookmarked Search Criterion
+---
+
+# IsBookmarked Criterion
+
+The [`IsBookmarked` Search Criterion](../../api/php_api/php_api_reference/classes/Ibexa-Contracts-Core-Repository-Values-Content-Query-Criterion-Location-IsBookmarked.html)
+searches for location based on whether it's bookmarked or not.
```
Comment (adriendupuis) [inline docs/search/criteria_reference/isbookmarked_criterion.md:8]: Bookmarked by anyone or bookmarked by the current user?

Comment (adriendupuis) [reply docs/search/criteria_reference/isbookmarked_criterion.md:8]: According to [its implementation](https://github.com/ibexa/core/blob/main/src/lib/Search/Legacy/Content/Location/Gateway/CriterionHandler/Location/IsBookmarked.php#L70), it's about the current user.

```suggestion
searches for location based on whether it's bookmarked or not by the current user.
```

File: docs/search/criteria_reference/isbookmarked_criterion.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+This Criterion is available only for location Search.
+It works with current user reference.
```
Comment (adriendupuis) [inline docs/search/criteria_reference/isbookmarked_criterion.md:11]: Oh, it there. May be a bit to late.

File: docs/search/criteria_reference/isbookmarked_criterion.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+## Arguments
+
+- `value` - bool representing whether to search for bookmarked location (default `true`) or not bookmarked location (`false`)
+
+## Example
+
+### PHP
+
+``` php
+[[= include_file('code_samples/search/location/isbookmarked_criterion.php') =]]
```
Comment (adriendupuis) [inline docs/search/criteria_reference/isbookmarked_criterion.md:22]: Just excluding the `<?php` file beginning so it looks like an extract of a larger code, like suggesting thatthe SearchService is injected somewhere above the sample.

```suggestion
[[= include_file('code_samples/search/location/isbookmarked_criterion.php', 2) =]]
```

**Review by adriendupuis — APPROVED** [review]
My suggestions are optional. It's already good.

**Pull request merged**